routes/auth.js

```diff
@@ -0,0 +1,75 @@
+const jwt = require("jsonwebtoken");
+const Joi = require('joi')
+const bcrypt = require("bcryptjs");
+const { User, validate } = require("../models/users");
+const mongoose = require("mongoose");
+const express = require("express");
+const router = express.Router();
+
+router.post("/register", async (req, res) => {
+  // Validates request body
+  const { error } = validate(req.body);
+  if (error) return res.status(400).send(error.details[0].message);
+
+  // Checks if user is already in the database
+  let user = await User.findOne({ email: req.body.email });
```

Comment:
Can we have all the logic part, which are in routes moved to `controllers/` so we can isolate the logic part and will make it easy to write unit tests?

Reply:
something like; all user related actions will go into `controllers/users.js` and we can have couple of methods like `createUser, findUser, updateUser and deleteUser` this also makes it easy for us to access user data if needed in other parts also.

Reply:
i second this! great suggestion @Kamalnrf

Reply:
@Kamalnrf and I pared program and talked about it last night. We'll probably do that for the other routes, but there isn't a need to pull out the logic in the Auth routes since those are the only places those methods will be used.